### PR TITLE
Refreshed look with new logos

### PR DESCRIPTION
**Pull request by chalin**
Updated:
- Logos, based on the new official Dart and Flutter logos.
- Some color adjustments
- Switched to using Roboto font, which is what the current Flutter site uses.

This PR doesn't include any frontpage redesign. That will be addressed separately (if at all for this site).

Contributes to #651 and #637

Staged at https://dartlang-org-staging-0.firebaseapp.com

cc @anders-sandholm @JekCharlsonYu 

---

- [x] Logos used in site pages.
- [x] favicon.ico
- [x] Social media shared images: Use new logo as og_image #832 
- [x] Ensure submodule branch was merged and that this PR points to the updated master.
- [x] Examples: any use of logos, including favicon.
- [x] Delete unused `src/twitter` folder and the images it contains? Cf. https://github.com/dart-lang/site-www/issues/651#issuecomment-386271055:
  - Except for (currently blocked) followup item: #833.

### CODE REVIEW

**Comment by chalin** [comment]
I used ImageMagick to create the favicon.ico, e.g.:

    convert angular/logo/square.png -define icon:auto-resize=128,64,48,32,16 angular/favicon.ico

**Comment by chalin** [comment]
Staging server refreshed (to include the updated favicon.ico).

**Comment by chalin** [comment]
Staging server refreshed with new social media sharing image, which (according to the Facebook Sharing Debugger), will appear like this:

screen shot 2018-05-02 at 17 51 36

**Comment by anders-sandholm** [comment]
I really like what I see. Thanks Patrice for getting all these beautiful changes lined up.
Unless there are obvious parts missing, let's get this landed.
Then we can always iterate further in the coming days.

**Comment by chalin** [comment]
The final submodule sync commit is in.
@kwalrath - awaiting for your official thumbs up.

**Comment by chalin** [comment]
I've scanned the entire site for any images that might contain old logos. All that remains is to update DartPad screenshots; see #833.

File: src/_assets/stylesheets/_variables.scss [hunk 1]
```diff
@@ -16,14 +16,14 @@ $blue-light:                #e2f4fd;
 $brand-primary:             #1155cc;
 
 // Blue color taken from gmail's stylesheet.
-$default-link:              #1155cc;
-$default-link-visited:      #1155cc;
+$default-link:              #0175C2;
```
Comment (kwalrath) [inline src/_assets/stylesheets/_variables.scss]: The comment above says this blue color is taken from gmail's stylesheet. Is that really true?

File: src/_assets/stylesheets/_variables.scss [hunk 2]
```diff
@@ -16,14 +16,14 @@ $blue-light:                #e2f4fd;
...
+$default-link-visited:      #0175C2;
 
 // Typeography
-$font-family-base:          'Source Sans Pro', sans-serif;
-$font-family-condensed:     'Trade Gothic W01', sans-serif;
-$font-family-extended:      'Trade Gothic W01', sans-serif;
-$font-family-serif:         'Trade Gothic W01', sans-serif;
+$font-family-base:          Roboto, sans-serif;
```
Comment (kwalrath) [inline src/_assets/stylesheets/_variables.scss]: While you're in here... Typeography -> Typography
:)

File: src/_includes/page-footer.html [hunk 3]
```diff
@@ -4,7 +4,7 @@
       <div class="col-sm-12 col-md-3">
         <div class="content">
           <div class="brand">
-            <img src="{% asset_path 'logo-white.svg' %}" alt="Dart logo" class="brand"/>
+            <img src="{% asset_path 'shared/dart/logo+text/horizontal/white.svg' %}" alt="Dart logo" class="brand"/>
```
Comment (kwalrath) [inline src/_includes/page-footer.html]: This logo seems less visible against the black footer than the old one, but I think we can fix this later (if at all).

**Comment by kwalrath** [comment]
I just noticed that the font is heavier than what I see on flutter.io. E.g. look at https://dartlang-org-staging-0.firebaseapp.com/guides/get-started vs. https://flutter.io/setup-macos/. It looks different from https://developers.google.com/style/, too (header font is different, and it looks like d.g.c uses an equally heavy but grayer font, which is pretty but questionable for accessibility reasons).

I'm OK with these font diffs — just pointing them out, in case anyone cares.

File: src/_assets/stylesheets/_variables.scss (hunk 1 above)
Comment (chalin) [reply src/_assets/stylesheets/_variables.scss]: Good catch. Removed.

**Comment by chalin** [comment]
Yes, the main font is scaled on flutter.io, but both sites use 16pt as the font base. I haven't found where in the flutter.io styling that they do the scaling. As you mention, we can address that later as necessary.

File: src/_assets/stylesheets/_variables.scss (hunk 1 above)
Comment (chalin) [reply src/_assets/stylesheets/_variables.scss]: Oops, I just noticed that I forgot to change `$brand-primary` to also be #0175C2. I've done that now. I've refreshed the staging server. PTAL (the color is very similar).

Comment (chalin) [reply src/_assets/stylesheets/_variables.scss]: (You'll likely have to empty the browser cache for the new CSS to load.)

**Pull request merged**